core/src/test/java/org/elasticsearch/index/query/BoolQueryBuilderTest.java

```diff
@@ -0,0 +1,136 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.MatchAllDocsQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.BooleanClause.Occur;
+import org.elasticsearch.common.lucene.search.Queries;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.elasticsearch.common.lucene.search.Queries.fixNegativeQueryIfNeeded;
+
+public class BoolQueryBuilderTest extends BaseQueryTestCase<BoolQueryBuilder> {
+
+    @Override
+    protected void assertLuceneQuery(BoolQueryBuilder queryBuilder, Query query, QueryParseContext context) {
+        if (queryBuilder.queryName() != null) {
+            Query namedQuery = context.copyNamedFilters().get(queryBuilder.queryName());
+            if (queryBuilder.hasClauses()) {
+                assertThat(namedQuery, equalTo(query));
+            } else {
+                assertNull(namedQuery);
+            }
+        }
+    }
+
+    @Override
+    protected BoolQueryBuilder createTestQueryBuilder() {
+        BoolQueryBuilder query = new BoolQueryBuilder();
+        if (randomBoolean()) {
+            query.boost(2.0f / randomIntBetween(1, 20));
+        }
+        if (randomBoolean()) {
+            query.adjustPureNegative(randomBoolean());
+        }
+        if (randomBoolean()) {
+            query.disableCoord(randomBoolean());
+        }
+        if (randomBoolean()) {
+            query.minimumNumberShouldMatch(randomIntBetween(1, 10));
+        }
+        for (int i = 0; i < randomIntBetween(0, 3); i++) {
+            query.must(createRandomQuery());
+        }
+        for (int i = 0; i < randomIntBetween(0, 3); i++) {
+            query.mustNot(createRandomQuery());
+        }
+        for (int i = 0; i < randomIntBetween(0, 3); i++) {
+            query.should(createRandomQuery());
+        }
+        for (int i = 0; i < randomIntBetween(0, 3); i++) {
+            query.filter(createRandomQuery());
+        }
+        if (randomBoolean()) {
+            query.queryName(randomUnicodeOfLengthBetween(3, 15));
+        }
+        return query;
+    }
+
+    private static QueryBuilder createRandomQuery() {
```

Comment:
I tried moving this to a separate class but since method uses randomness, its better situated inside a test (or make that new Util-class a test, which feels strange to me). So I moved it to the BaseQueryTestCase where it can be re-used by other Query tests.